plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +721,57 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+void normalizationFormCheck(UnicodeString & source, const char * form)
+{
+    UErrorCode errorCode = U_ZERO_ERROR;
+    if (form[2] == 'C')
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfc", UNormalization2Mode::UNORM2_DECOMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    else if (form[2] == 'D')
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfc", UNormalization2Mode::UNORM2_DECOMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    else if (form[3] == 'C')
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfkc", UNormalization2Mode::UNORM2_COMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    else if (form[3] == 'D')
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfkc", UNormalization2Mode::UNORM2_DECOMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+}
+
+void removeSuffix(UnicodeString & source, UnicodeString const & suf)
+{
+    if (source.isEmpty() || suf.isEmpty())
+    {
+        return;
+    }
+    int32_t last = source.length();
+    int32_t suffixLength = suf.length();
+    source.removeBetween((last - suffixLength), last);
+}
+
+
+bool endsWith(UnicodeString & processed, UnicodeString & suffix)
+{
+    if (processed.isEmpty() || suffix.isEmpty())
+    {
+        return false;
+    }
+    suffix.trim();
```

Comment:
Cleaner to trim the string before calling this function (having an apparently pure function with side effects on its parameters is scope for confusion).  Parameters should then become const references.